zkevm-circuits/src/evm_circuit/execution/block_ctx.rs

```diff
@@ -188,17 +196,20 @@ impl<F: Field> ExecutionGadget<F> for BlockCtxU256Gadget<F> {
     }
 }
 
+#[cfg(feature = "scroll")]
 #[derive(Clone, Debug)]
-pub(crate) struct DifficulityGadget<F> {
+pub(crate) struct DifficultyGadget<F> {
     same_context: SameContextGadget<F>,
 }
 
-impl<F: Field> ExecutionGadget<F> for DifficulityGadget<F> {
-    const NAME: &'static str = "DIFFICULITY";
+#[cfg(feature = "scroll")]
+impl<F: Field> ExecutionGadget<F> for DifficultyGadget<F> {
+    const NAME: &'static str = "DIFFICULTY";
 
-    const EXECUTION_STATE: ExecutionState = ExecutionState::BLOCKCTXU256;
+    const EXECUTION_STATE: ExecutionState = ExecutionState::DIFFICULTY;
```

Comment:
`ExecutionState::Difficulty` is only constrained by the `DifficultyGadget` when `scroll` feature is enabled, which means if we **disable** the `scroll` feature, then this state is **not constrained at all** (this is a big soundness issue).

Reply:
Fixed in commit [a511ed4](https://github.com/scroll-tech/zkevm-circuits/pull/1033/commits/a511ed4157341a9555ef28297914000a935287c5).